templates/deployment.yaml

```diff
@@ -108,6 +108,15 @@ spec:
           {{- end}}
         {{- end}}
       {{- end}}
+      {{- if .EnableTopologySpreadConstraints}}
+      topologySpreadConstraints:
+        - maxSkew: 1
+          topologyKey: kubernetes.io/hostname
+          whenUnsatisfiable: ScheduleAnyway
```

Comment:
I see that
-  the default value for this should be DoNotSchedule. 
- For pod antifinity part, I think it should be removed as it is doing the same thing.

Reply:
@afarid for now this behaviour will not be default for all applications. We have many apps with > 100 replicas and we don’t want them to spin up and run on 100 nodes.. Affinity and topologysprad constraints are additive in nature and both can co exists

Reply:
The topologySpreadConstraints is not functioning correctly with Node Affinity. For instance, if there are 200 pods and two nodes, the rule ensures that pods are evenly distributed across the nodes. 

The conclusion is,  The cluster will not scale unless two conditions are met:

No Capacity: There is insufficient capacity on the nodes.
MaxSkew can't be satisfied: The maximum skew constraint is violated.

setting to DoNotSchedule  ensures that pods are not scheduled within the same failure domain.